### ISSUE TITLE
does this already exist

### ISSUE THREAD

**Issue by mpadge** [mpadge/stellar#1]
ahoy @maelle @hrbrmstr @jonocarroll and anybody else you care to inform: Lazy research here, but I can't find any current way to search my github stars like [`astral`](https://github.com/astralapp/astral) can (click [here](https://astralapp.com/) if you're unfamilar). This is somewhat different from the intention of [`starryeyes`](https://github.com/jonocarroll/starryeyes) - although @jonocarroll, more than happy to just expand your idea with this if you think that might somehow fit. Before I embark on this, do I err in thinking this hasn't been done before? And if not - anybody keen to help?

Stars can be extracted directly from the gh API, then a second query for repo description texts which can then simply be searched (fancy-pants tokenized and that guff too). The vision is a screen dump of the top 9 hits, type a number for instant opening of web page, other stuff to either go further or just stop.

**Comment by hrbrmstr** [comment]
(to help me clarify) search — programmatically or CLI — by repo name / title /content?

**Comment by hrbrmstr** [comment]
nothing is coming to mind. but, now that you've mentioned it, being able to wrap something in R into a `bash` alias to do `$ stargrep <terms>` would be all kinds of cool.

I volunteer to help with the CLI interface :-)

**Comment by mpadge** [comment]
yeah, definitely all CLI in my vision o'things - pretty much [`flipper`](https://github.com/ropenscilabs/flipper) for gh stars. Probably initially just repo name and description, but with API4, I guess you get more stuff per call, so some content too perhaps?

**Comment by jonocarroll** [comment]
This (https://github.com/Sliim/helm-github-stars) is the only thing that comes to mind, but yeah - the idea is currently different to what's already in starryeyes. There I'm grabbing a feed and looking for who (other people the user follows) starred what. 

I'd be very happy to collaborate on adding 'what has a **specific user** starred?' summary/search into that project. Once you have a cli API for getting the info, showing a shiny representation of it can also be neat. 

Say the word and I'll add you as a collaborator.

**Comment by jonocarroll** [comment]
@hrbrmstr with `utils::Rscript` or `littler`, that should be very do-able these days.

**Comment by maelle** [comment]
Very cool. I do not know of sthg similar

**Comment by hrbrmstr** [comment]
I gotta add some _sweet_ `cli` and `crayon` crunchy goodness to it, tho, @jonocarroll

**Comment by jonocarroll** [comment]
I _presume_ that works in a bash script calling R (in a supporting terminal) but I definitely need to find out.

I'll invite @mpadge and @hrbrmstr to starryeyes just in case. Do as you like.

**Comment by mpadge** [comment]
@jonocarroll thanks for the feedback and invite! The two ideas seem pretty perfectly complementary - I'm envisioning a pure CLI thing coz I don't like leaving my terminal, and one that is primarily text-search based, but now with extra ideas sketched out in current `README` thanks to your inspiration! `starryeyes` is then a shiny embellishment for keeping up to date. How about we develop both as separate instances first, then talk about merging? I don't think there be too much overlap in the initial phases at least, but do think that an ultimate merrge would likely generate a pretty useful package

**Comment by jonocarroll** [comment]
As you've already noticed, my implementation of internals is out of date. I'm happy to help add some (better?) functions to process the feed in this repo if that's better. The two ideas chain together nicely as a discovery tool -- find a user's star list, choose a repo they've starred, find which repos that dev starred, down the rabbit hole you go. Starting from your own (via a processing of your feed) is just one possibility. It would also tie into a "which of this dev's other repos have I starred?" search/tabulation/summary. Anything shiny requires a fully capable cli API first (IMHO), so I'd definitely build that technology first, then ice the cake.

The 'extract a user's stars', 'launch a browser to this repo', and general GitHub access is all functionality I either envisage or could have done better so I'm more than happy to move the feed parsing (and shiny) part over here. starryeyes has all of 4 stars itself (apart from my own) and 0 forks, so I'm not leaving a lot behind. I won't tread on any toes though if you're keen to tackle any or all of these bits yourself. The offer's there if you want it.

tl;dr -- this is a great idea, and there's lots to be done in one or both repos. I'm looking forward to it!

**Comment by mpadge** [comment]
Awesome! Let's do it then, and I'll move this to the neutral, fertile ground of `ropenscilabs` and off we go. @maelle This might also offer a good starting place for pulling some of your [`ghrecipes`](https://github.com/ropenscilabs/ghrecipes) into shape by giving a narrower focus that will more quickly work towards a useable, releasable package
